### PR TITLE
[com_finder] index view: searchtools ordering, bug corrections and some improvements

### PR DESCRIPTION
**Pull request by andrepereiradasilva**
#### Summary of Changes

This PR normalizes this view by adding the ordering to the searchtools in com_finder filters view. 

Also:
- Corrects some bugs in the searchtools (for instance, on select the state filter is not selected, ordering by type is wrong).
- Some code style improvements.
- Some code improvements.
- Normalizes layout with other views.
- Moves messages to the view.html file.
- Improves the mobile/tablet views.
###### Before PR

image
###### After PR

image
#### Testing Instructions
1. Use latest staging and apply patch
2. Go to Components -> Smart Search
3. Press Index
4. Test if search, filters, ordering and page limit (aka searchtools) are working properly.
5. Test in mobile and tablet views.

### CODE REVIEW

**Comment by brianteeman** [comment]
Presumably there was a specific reason for the full non-sef url being presented @chrisdavenport

**Comment by mbabker** [comment]
Generating front-end SEF URLs for display in the admin involves some trickery.  You can't use `JRoute::_()` to do it because of the coupling to `JFactory::getApplication()`.  This method will do the trick if you really wanted to show SEF URLs:

```php
/**
 * Replacement for JRoute::_() for routing from Joomla's backend
 *
 * @param   string  $url  The URL to process
 *
 * @return  string  The routed URL
 */
public static function getFrontendRoute($url)
{
	// Get the router.
	$router = JApplicationCms::getInstance('site')->getRouter();

	// Make sure that we have our router
	if (!$router)
	{
		return null;
	}

	if ((strpos($url, '&') !== 0) && (strpos($url, 'index.php') !== 0))
	{
		return $url;
	}

	// Build route.
	$uri = $router->build($url);
	$url = $uri->toString(['path', 'query', 'fragment']);

	// Replace spaces.
	$url = preg_replace('/\s/u', '%20', $url);

	// Replace '/administrator'
	$url = str_replace('/administrator', '', $url);

	$url = htmlspecialchars($url);

	return $url;
}
```

**Comment by andrepereiradasilva** [comment]
@mbabker yeah i notice that.

Could the php 5.3 version of that method be in https://github.com/joomla/joomla-cms/blob/staging/libraries/cms/router/administrator.php?

File: administrator/components/com_finder/models/forms/filter_index.xml [hunk 1]
```diff
@@ -1,43 +1,60 @@
...
-                description="COM_FINDER_LIST_LIMIT_DESC"
-                onchange="this.form.submit();"
-                />
-    </fields>
+	<fields name="filter">
+		<field
+			name="search"
+			type="text"
+			label="COM_FINDER_INDEX_SEARCH_DESC"
+			description="COM_FINDER_INDEX_SEARCH_DESC"
```
Comment (brianteeman) [inline administrator/components/com_finder/models/forms/filter_index.xml]: Shouldnt there be separate strings for the Label and Description

Comment (brianteeman) [inline administrator/components/com_finder/models/forms/filter_index.xml]: Not sure I understand why it is changed from COM_FINDER_INDEX_SEARCH_DESC to COM_FINDER_FILTER_SEARCH_DESC"

Comment (andrepereiradasilva) [inline administrator/components/com_finder/models/forms/filter_index.xml]: what does the label, in this case, serves for?

Comment (brianteeman) [inline administrator/components/com_finder/models/forms/filter_index.xml]: No idea - you are the one that made the change :)

Comment (andrepereiradasilva) [inline administrator/components/com_finder/models/forms/filter_index.xml]: you got me with that one! :smile: 

The things is, in the current search field of the search tools:
- `label=` that i know of, is not used at all
- `description=` is for the tooltip of the search field
- `hint=` is for the placeholder of the search field

but you're right, it should be _LABEL there.  i will change it

Comment (brianteeman) [inline administrator/components/com_finder/models/forms/filter_index.xml]: Isnt the label required for accessibility on all fields?

Comment (andrepereiradasilva) [inline administrator/components/com_finder/models/forms/filter_index.xml]: i think so, maybe it's hidden let em check

Comment (andrepereiradasilva) [inline administrator/components/com_finder/models/forms/filter_index.xml]: there is an hidden label, but is not using the label attribute it seems...

Comment (andrepereiradasilva) [inline administrator/components/com_finder/models/forms/filter_index.xml]: as you can see here https://github.com/joomla/joomla-cms/blob/staging/layouts/joomla/searchtools/default/bar.php#L34
currently the label is hardcoded as "JSEARCH_FILTER" for all search fields in searchtools.

Anyway i change it to _LABEL.

**Comment by chrisdavenport** [comment]
> Presumably there was a specific reason for the full non-sef url being presented @chrisdavenport

Both SEF and non-SEF URLs would be useful during debugging.  Since this component originates from something that was originally written for Joomla 1.5, I suspect that the non-SEF link was shown mainly because the SEF link was hard to reliably generate.

**Comment by chrisdavenport** [comment]
@andrepereiradasilva While you're messing with this component, there is a small issue that I noticed a while back that you can probably fix at the same time:

If you use Search Tools to make some selection which results in an empty list, you will see this message: "No content has been indexed. Start the indexer by pressing the Index button in the toolbar." which is actually wrong in that context.

**Comment by mbabker** [comment]
> Could the php 5.3 version of that method be in https://github.com/joomla/joomla-cms/blob/staging/libraries/cms/router/administrator.php?

I'd throw it in `JRoute` instead of `JRouterAdministrator` personally.  I'd also battle test it a bit harder than I did with my hack (if you ever get a frontend path with `/administrator` in it that'll get string replaced and it doesn't cope with the HTTP(S) detection like `JRoute::_()` does.  Otherwise, it's GPL code ripped off from a GPL project, do as you please 😉

**Comment by andrepereiradasilva** [comment]
@chrisdavenport, i solved that now. this is how it looks after the latest commit

Withour index

image

With index

image


@mbabker thanks, i will check that later

**Comment by chrisdavenport** [comment]
I have tested this item :white_check_mark: successfully on f56c2d6580884b79bda89e0b7290498759edb22e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10049">issues.joomla.org/joomla-cms/10049</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on f56c2d6580884b79bda89e0b7290498759edb22e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10049">issues.joomla.org/joomla-cms/10049</a>.</sub>

**Comment by brianteeman** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10049">issues.joomla.org/joomla-cms/10049</a>.</sub>

**Pull request merged**